src/main/java/org/jabref/gui/LibraryTab.java

```diff
@@ -1197,4 +1205,26 @@ public String toString() {
     public LibraryTabContainer getLibraryTabContainer() {
         return tabContainer;
     }
+
+    /**
+     * Takes an entry and increments the number of times it has been viewed by 1.
+     */
+    private void incrementViewCount(BibEntry entry) {
+        MVStore mvStore = PopularityGroup.getMVStore();
+        synchronized (mvStore) {
+            MVMap<String, Integer> viewCounts = mvStore.openMap("entryViewCounts");
+            MVMap<String, Long> lastViewTimestamps = mvStore.openMap("lastViewTimestamps");
+
+            long currentTime = System.currentTimeMillis();
+            String uniqueKey = PopularityGroup.getUniqueKeyForEntry(entry);
+            long lastViewTime = lastViewTimestamps.getOrDefault(uniqueKey, 0L);
+
+            if (currentTime - lastViewTime >= ONE_HOUR_IN_MILLISECONDS && EntryTabViewModel.trackViewsProperty().get()) {
```

Comment:
The check for the property is too late - everything doesn't need to be done if the property is false.